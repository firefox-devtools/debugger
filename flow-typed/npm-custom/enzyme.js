/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// Using this custom definition of the enzyme module avoids generating some
// flow errors that are unrelated to uses of enzyme itself...
// It would be nice to clean this up.
declare module "enzyme" {
  declare module.exports: any;
}
