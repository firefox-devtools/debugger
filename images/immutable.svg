<!-- This Source Code Form is subject to the terms of the Mozilla Public
   - License, v. 2.0. If a copy of the MPL was not distributed with this
   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 20010904//EN"
 "http://www.w3.org/TR/2001/REC-SVG-20010904/DTD/svg10.dtd">
<svg version="1.0" xmlns="http://www.w3.org/2000/svg"
 width="200.000000pt" height="200.000000pt" viewBox="0 0 200.000000 200.000000"
 preserveAspectRatio="xMidYMid meet">

<g transform="translate(0.000000,200.000000) scale(0.100000,-0.100000)"
fill="#000000" stroke="none">
<path d="M247 1663 c-4 -3 -7 -235 -7 -515 l0 -508 175 0 175 0 0 515 0 515
-168 0 c-93 0 -172 -3 -175 -7z m183 -508 c0 -336 -2 -385 -15 -385 -13 0 -15
49 -15 385 0 336 2 385 15 385 13 0 15 -49 15 -385z"/>
<path d="M697 1664 c-4 -4 -7 -236 -7 -516 l0 -508 165 0 165 0 2 181 3 181
99 -146 c54 -80 102 -146 106 -146 3 0 51 66 106 148 l99 147 3 -182 2 -183
160 0 160 0 0 515 0 515 -132 0 -133 -1 -130 -194 c-71 -107 -132 -194 -135
-194 -3 -1 -64 86 -135 192 l-131 192 -130 3 c-71 1 -133 -1 -137 -4z m356
-374 c92 -138 172 -250 176 -250 4 0 84 112 177 250 112 166 175 250 187 250
16 0 17 -25 17 -385 0 -336 -2 -385 -15 -385 -13 0 -15 46 -15 350 0 193 -4
350 -8 350 -4 0 -82 -108 -172 -240 -90 -132 -167 -240 -171 -240 -4 0 -81
108 -171 240 -90 131 -166 239 -170 240 -5 0 -8 -157 -8 -350 0 -304 -2 -350
-15 -350 -13 0 -15 49 -15 385 0 360 1 385 18 385 11 -1 74 -86 185 -250z"/>
</g>
</svg>
