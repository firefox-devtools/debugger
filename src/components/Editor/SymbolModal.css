.symbol-modal {
  position: absolute;
  left: 16.5%;
  z-index: 10;
  width: 66.66%;
  height: 230px;
  background-color: var(--theme-codemirror-gutter-background);
  box-shadow: 2px 4px 6px #dde1e4;
}

.symbol-modal .input-wrapper {
  display: flex;
  justify-content: center;
}

.symbol-modal .search-field {
  height: var(--editor-searchbar-height);
}

.symbol-modal .close-btn {
  padding: 6px;
}
