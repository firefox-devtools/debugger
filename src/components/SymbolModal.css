.symbol-modal-wrapper {
  position: fixed;
  width: 100%;
  height: 100%;
  z-index: 9;
}

.symbol-modal {
  position: absolute;
  left: calc(50% - 250px);
  z-index: 10;
  width: 500px;
  height: 230px;
  background-color: var(--theme-codemirror-gutter-background);
  box-shadow: 2px 4px 6px var(--popup-shadow-color);
  top: 30px;
}

.theme-dark .symbol-modal {
  box-shadow: 2px 4px 6px var(--popup-shadow-color);
}

.symbol-modal .input-wrapper {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.symbol-modal .close-btn {
  padding: 6px;
}

.symbol-modal .result-list {
  height: calc(100% - 28px);
  overflow-y: auto;
}

@media (max-width: 520px) {
  .symbol-modal {
    width: 80%;
    left: 10%;
  }
}
