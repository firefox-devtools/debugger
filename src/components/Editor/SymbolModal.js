// @flow

import { DOM as dom, createFactory, Component, PropTypes } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { filter } from "fuzzaldrin-plus";
import {
  getActiveSearchState,
  getSymbolSearchType,
  getSelectedSource,
  getSymbols,
  getSymbolSearchResults,
  getSymbolSearchQueryState
} from "../../selectors";
import actions from "../../actions";

import { scrollList } from "../../utils/result-list";

import _SearchInput from "../shared/SearchInput";
const SearchInput = createFactory(_SearchInput);

import _ResultList from "../shared/ResultList";
const ResultList = createFactory(_ResultList);

import type { ActiveSearchType, SymbolSearchType } from "../../reducers/ui";
import type {
  SymbolDeclaration,
  SymbolDeclarations
} from "../../utils/parser/getSymbols";

import type { Location as BabelLocation } from "babel-traverse";
import type { SourceRecord } from "../../reducers/sources";
import type { SelectSourceOptions } from "../../actions/sources";

export type FormattedSymbolDeclaration = {
  id: string,
  title: string,
  subtitle: string,
  value: string,
  location: BabelLocation,
  parameterNames?: string[]
};

function formatSymbol(symbol: SymbolDeclaration): FormattedSymbolDeclaration {
  return {
    id: `${symbol.name}:${symbol.location.start.line}`,
    title: symbol.name,
    subtitle: `:${symbol.location.start.line}`,
    value: symbol.name,
    location: symbol.location
  };
}

type SymbolModalState = { selectedResultIndex: number };

import "./SymbolModal.css";

class SymbolModal extends Component {
  state: SymbolModalState;

  props: {
    enabled: boolean,
    query: string,
    selectSource: (string, ?SelectSourceOptions) => any,
    selectedSource?: SourceRecord,
    symbols: SymbolDeclarations,
    symbolType: SymbolSearchType,
    setSelectedSymbolType: SymbolSearchType => any,
    toggleActiveSearch: (?ActiveSearchType) => any,
    setSymbolSearchQuery: string => any,
    updateSymbolSearchResults: any,
    highlightLineRange: ({ start: number, end: number }) => void,
    clearHighlightLineRange: () => void,
    symbolSearchResults: any
  };

  constructor(props) {
    super(props);
    this.state = { selectedResultIndex: 0 };

    const self: any = this;
    self.onEscape = this.onEscape.bind(this);
    self.closeModal = this.closeModal.bind(this);
    self.onChange = this.onChange.bind(this);
    self.onKeyUp = this.onKeyUp.bind(this);
    self.updateResults = this.updateResults.bind(this);
    self.traverseResults = this.traverseResults.bind(this);
    self.renderResults = this.renderResults.bind(this);
    self.buildSummaryMsg = this.buildSummaryMsg.bind(this);
    self.buildPlaceHolder = this.buildPlaceHolder.bind(this);
  }

  componentWillUnmount() {
    const shortcuts = this.context.shortcuts;
    shortcuts.off("Escape");
    shortcuts.off(L10N.getStr("symbolSearch.search.key2"));
  }

  componentDidMount() {
    const shortcuts = this.context.shortcuts;
    shortcuts.on("Escape", (_, e) => this.onEscape(e));
    shortcuts.on(L10N.getStr("symbolSearch.search.key2"), (_, e) => {
      this.props.toggleActiveSearch("symbol");
      this.props.setSelectedSymbolType("functions");
    });
  }

  componentDidUpdate(prevProps: any, prevState: any) {
    if (this.refs.resultList && this.refs.resultList.refs) {
      scrollList(this.refs.resultList.refs, this.state.selectedResultIndex);
    }
  }

  onEscape(e: SyntheticEvent) {
    this.props.toggleActiveSearch();
  }

  onChange(e: SyntheticInputEvent) {
    const {
      selectedSource,
      setSymbolSearchQuery,
      setSelectedSymbolType
    } = this.props;
    if (!selectedSource || !selectedSource.get("text")) {
      return;
    }

    if (e.target.value[0] === "@") {
      setSelectedSymbolType("functions");
    } else if (e.target.value[0] === "#") {
      setSelectedSymbolType("variables");
    }

    setSymbolSearchQuery(e.target.value);

    return this.updateResults(e.target.value.slice(1));
  }

  closeModal() {
    this.props.toggleActiveSearch();
    this.props.clearHighlightLineRange();
  }

  selectResultItem(e: SyntheticEvent, item: SymbolDeclaration) {
    const { selectSource, selectedSource } = this.props;

    if (!selectedSource) {
      return;
    }
    selectSource(selectedSource.get("id"), {
      line: item.location.start.line
    });

    this.closeModal();
  }

  updateResults(query: string) {
    const {
      selectedSource,
      updateSymbolSearchResults,
      symbolType,
      symbols
    } = this.props;

    if (!query || !selectedSource) {
      return;
    }

    const symbolSearchResults = filter(symbols[symbolType], query, {
      key: "value"
    });

    updateSymbolSearchResults(symbolSearchResults);
  }

  traverseResults(direction: number) {
    const { selectedResultIndex } = this.state;
    const { symbolSearchResults } = this.props;
    const searchResults = symbolSearchResults;
    const resultCount = searchResults.length;
    const index = selectedResultIndex + direction;
    const nextIndex = (index + resultCount) % resultCount;

    this.setState({ selectedResultIndex: nextIndex });
    this.onSelectResultItem(searchResults[nextIndex]);
  }

  onSelectResultItem(item: FormattedSymbolDeclaration) {
    const {
      selectSource,
      selectedSource,
      symbolType,
      highlightLineRange
    } = this.props;

    if (selectedSource && symbolType !== "functions") {
      selectSource(selectedSource.get("id"), {
        line: item.location.start.line
      });
    }

    if (selectedSource && symbolType === "functions") {
      highlightLineRange({
        start: item.location.start.line,
        end: item.location.end.line,
        sourceId: selectedSource.get("id")
      });
    }
  }

  onKeyUp(e: SyntheticKeyboardEvent) {
    e.preventDefault();
    const { symbolSearchResults, enabled, query } = this.props;
    if (!enabled || query == "") {
      return;
    }

    const searchResults = symbolSearchResults;

    if (e.key === "ArrowUp") {
      this.traverseResults(-1);
    } else if (e.key === "ArrowDown") {
      this.traverseResults(1);
    } else if (e.key === "Enter") {
      if (searchResults.length) {
        this.selectResultItem(e, searchResults[this.state.selectedResultIndex]);
      }
      this.closeModal();
    } else if (e.key === "Tab") {
      this.closeModal();
    }
  }

  renderResults() {
    const { symbolSearchResults } = this.props;
    const { selectedResultIndex } = this.state;

    const { query, enabled } = this.props;
    if (!query || !enabled || !symbolSearchResults.length) {
      return;
    }

    return ResultList({
      items: symbolSearchResults,
      selected: selectedResultIndex,
      selectItem: this.selectResultItem,
      ref: "resultList"
    });
  }

  buildSummaryMsg() {
    const { selectedResultIndex } = this.state;
    const count = this.resultsCount();

    if (count > 1) {
      return L10N.getFormatStr(
        "editor.searchResults",
        selectedResultIndex + 1,
        count
      );
    } else if (count === 1) {
      return L10N.getFormatStr("editor.singleResult");
    }
  }

  resultsCount() {
    return this.props.symbolSearchResults.length;
  }

  buildPlaceHolder() {
    const { symbolType } = this.props;
    return L10N.getFormatStr(`symbolSearch.search.${symbolType}Placeholder`);
  }

  render() {
    const { enabled, query } = this.props;
    if (!enabled) {
      return dom.div();
    }
    return dom.div(
      { className: "symbol-modal" },
      dom.div(
        { className: "input-wrapper" },
        SearchInput({
          query,
          count: this.resultsCount(),
          placeholder: this.buildPlaceHolder(),
          summaryMsg: this.buildSummaryMsg(),
          onChange: this.onChange,
          onKeyUp: this.onKeyUp,
          handleNext: () => this.traverseResults(1),
          handlePrev: () => this.traverseResults(-1),
          handleClose: this.closeModal
        })
      ),
      this.renderResults()
    );
  }
}

SymbolModal.displayName = "SymbolModal";
SymbolModal.contextTypes = {
  shortcuts: PropTypes.object
};

function _getFormattedSymbols(state) {
  const source = getSelectedSource(state);
  if (!source) {
    return { variables: [], functions: [] };
  }
  const { variables, functions } = getSymbols(state, source.toJS());

  return {
    variables: variables.map(formatSymbol),
    functions: functions.map(formatSymbol)
  };
}

export default connect(
  state => ({
    query: getSymbolSearchQueryState(state),
    enabled: getActiveSearchState(state) === "symbol",
    symbolType: getSymbolSearchType(state),
    symbols: _getFormattedSymbols(state),
    symbolSearchResults: getSymbolSearchResults(state)
  }),
  dispatch => bindActionCreators(actions, dispatch)
)(SymbolModal);
