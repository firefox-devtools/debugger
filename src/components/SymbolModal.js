// @flow

import React, { Component, PropTypes } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { filter } from "fuzzaldrin-plus";
import {
  getActiveSearch,
  getSymbolSearchType,
  getSelectedSource,
  getSymbols
} from "../selectors";
import actions from "../actions";

import { scrollList } from "../utils/result-list";

import Modal from "./shared/Modal";

import SearchInput from "./shared/SearchInput";

import ResultList from "./shared/ResultList";

import type { ActiveSearchType, SymbolSearchType } from "../reducers/ui";
import type { SymbolDeclaration } from "../utils/parser/types";

import type { Location as BabelLocation } from "babel-traverse";
import type { SourceRecord } from "../reducers/sources";
import type { SelectSourceOptions } from "../actions/sources";

export type FormattedSymbolDeclaration = {
  id: string,
  title: string,
  subtitle: string,
  value: string,
  location: BabelLocation,
  parameterNames?: string[]
};

export type FormattedSymbolDeclarations = {
  variables: Array<FormattedSymbolDeclaration>,
  functions: Array<FormattedSymbolDeclaration>
};

function formatSymbol(symbol: SymbolDeclaration): FormattedSymbolDeclaration {
  return {
    id: `${symbol.name}:${symbol.location.start.line}`,
    title: symbol.name,
    subtitle: `:${symbol.location.start.line}`,
    value: symbol.name,
    location: symbol.location
  };
}

type SymbolModalState = {
  resultsIndex: number,
  results: ?Array<FormattedSymbolDeclaration>,
  query: ?string
};

import "./SymbolModal.css";

class SymbolModal extends Component {
  state: SymbolModalState;

  props: {
    enabled: boolean,
    selectSource: (string, ?SelectSourceOptions) => void,
    selectedSource?: SourceRecord,
    symbols: FormattedSymbolDeclarations,
    symbolType: SymbolSearchType,
    setActiveSearch: (?ActiveSearchType) => void,
    closeActiveSearch: () => void,
    highlightLineRange: ({ start: number, end: number }) => void,
    clearHighlightLineRange: () => void,
    symbolSearchResults: any
  };

  constructor(props) {
    super(props);
    this.state = { results: null, query: "", resultsIndex: 0 };

    const self: any = this;
    self.onClick = this.onClick.bind(this);
    self.closeModal = this.closeModal.bind(this);
    self.onChange = this.onChange.bind(this);
    self.onKeyUp = this.onKeyUp.bind(this);
    self.updateResults = this.updateResults.bind(this);
    self.traverseResults = this.traverseResults.bind(this);
    self.renderResults = this.renderResults.bind(this);
    self.buildSummaryMsg = this.buildSummaryMsg.bind(this);
    self.buildPlaceHolder = this.buildPlaceHolder.bind(this);
    self.selectResultItem = this.selectResultItem.bind(this);
  }

  componentDidMount() {
    this.updateResults(this.state.query);
  }

  componentDidUpdate(prevProps: any, prevState: any) {
    if (this.refs.resultList && this.refs.resultList.refs) {
      scrollList(this.refs.resultList.refs, this.state.resultsIndex);
    }

    if (!prevProps.enabled && this.props.enabled) {
      this.updateResults(this.state.query);
    }
  }

  onClick(e: SyntheticEvent) {
    e.stopPropagation();
  }

  onChange(e: SyntheticInputEvent) {
    const { selectedSource } = this.props;
    if (!selectedSource || !selectedSource.get("text")) {
      return;
    }

    this.setState({ query: e.target.value });
    return this.updateResults(e.target.value);
  }

  closeModal() {
    this.props.closeActiveSearch();
    this.props.clearHighlightLineRange();
  }

  selectResultItem(e: SyntheticEvent, item: ?FormattedSymbolDeclaration) {
    const { selectSource, selectedSource } = this.props;

    if (!selectedSource || !item) {
      return;
    }

    selectSource(selectedSource.get("id"), {
      line: item.location.start.line
    });

    this.closeModal();
  }

  updateResults(query) {
    const { symbolType, symbols } = this.props;

    let symbolSearchResults = symbols[symbolType];
    if (query == "") {
      this.setState({ results: symbolSearchResults });
      return;
    }

    symbolSearchResults = filter(symbolSearchResults, query, {
      key: "value"
    });

    this.setState({ results: symbolSearchResults });
  }

  traverseResults(direction: number) {
    const { resultsIndex, results } = this.state;
    const resultCount = this.resultsCount();
    const index = resultsIndex + direction;
    const nextIndex = (index + resultCount) % resultCount;

    this.setState({ resultsIndex: nextIndex });

    if (results) {
      this.onSelectResultItem(results[nextIndex]);
    }
  }

  onSelectResultItem(item: FormattedSymbolDeclaration) {
    const {
      selectSource,
      selectedSource,
      symbolType,
      highlightLineRange
    } = this.props;

    if (selectedSource && symbolType !== "functions") {
      selectSource(selectedSource.get("id"), {
        line: item.location.start.line
      });
    }

    if (selectedSource && symbolType === "functions") {
      highlightLineRange({
        start: item.location.start.line,
        end: item.location.end.line,
        sourceId: selectedSource.get("id")
      });
    }
  }

  onKeyUp(e: SyntheticKeyboardEvent) {
    e.preventDefault();
    const { enabled } = this.props;
    const { results, resultsIndex } = this.state;

    if (!enabled || !results) {
      return;
    }

    if (e.key === "ArrowUp") {
      this.traverseResults(-1);
    } else if (e.key === "ArrowDown") {
      this.traverseResults(1);
    } else if (e.key === "Enter") {
      this.selectResultItem(e, results[resultsIndex]);
      this.closeModal();
    } else if (e.key === "Tab") {
      this.closeModal();
    }
  }

  renderResults() {
    const { resultsIndex, results } = this.state;

    const { enabled } = this.props;
    if (!enabled || !results) {
      return null;
    }

    return (
      <ResultList
        key="results"
        items={results}
        selected={resultsIndex}
        selectItem={this.selectResultItem}
        ref="resultList"
      />
    );
  }

  renderInput() {
    const { query } = this.state;

    return (
      <div key="input" className="input-wrapper">
        <SearchInput
          query={query}
          count={this.resultsCount()}
          placeholder={this.buildPlaceHolder()}
          summaryMsg={this.buildSummaryMsg()}
          onChange={this.onChange}
          onKeyUp={this.onKeyUp}
          handleNext={() => this.traverseResults(1)}
          handlePrev={() => this.traverseResults(-1)}
          handleClose={this.closeModal}
        />
      </div>
    );
  }

  buildSummaryMsg() {
    const { resultsIndex } = this.state;
    const count = this.resultsCount();

    if (count > 1) {
      return L10N.getFormatStr("editor.searchResults", resultsIndex + 1, count);
    } else if (count === 1) {
      return L10N.getFormatStr("editor.singleResult");
    }
  }

  resultsCount() {
    return this.state.results ? this.state.results.length : 0;
  }

  buildPlaceHolder() {
    const { symbolType } = this.props;
    return L10N.getFormatStr(`symbolSearch.search.${symbolType}Placeholder`);
  }

  render() {
    const { enabled } = this.props;

    if (!enabled) {
      return null;
    }

    return (
      <Modal in={enabled} handleClose={this.closeModal}>
        {this.renderInput()}
        {this.renderResults()}
      </Modal>
    );
  }
}

SymbolModal.displayName = "SymbolModal";
SymbolModal.contextTypes = {
  shortcuts: PropTypes.object
};

function _getFormattedSymbols(state, source) {
  if (!source) {
    return { variables: [], functions: [] };
  }

  const { variables, functions } = getSymbols(state, source.toJS());

  return {
    variables: variables.map(formatSymbol),
    functions: functions.map(formatSymbol)
  };
}

export default connect(
  state => {
    const source = getSelectedSource(state);
    return {
      enabled: Boolean(getActiveSearch(state) === "symbol" && source),
      symbolType: getSymbolSearchType(state),
      symbols: _getFormattedSymbols(state, source)
    };
  },
  dispatch => bindActionCreators(actions, dispatch)
)(SymbolModal);
