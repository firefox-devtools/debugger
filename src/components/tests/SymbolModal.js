import React from "react";
import { shallow } from "enzyme";
import SymbolModal from "../SymbolModal";
import * as I from "immutable";

const SymbolModalComponent = SymbolModal.WrappedComponent;

function generateDefaults() {
  return {
    enabled: true,
    symbolType: "functions",
    selectedSource: I.Map({}),
    setSelectedSymbolType: () => {}
  };
}

function render(overrides = {}) {
  const defaults = generateDefaults();
  const props = { ...defaults, ...overrides };
  const component = shallow(<SymbolModalComponent {...props} />);
  return { component, props };
}

describe("SearchBar", () => {
  it("should render", () => {
    const { component } = render();
    expect(component).toMatchSnapshot();
  });
});
