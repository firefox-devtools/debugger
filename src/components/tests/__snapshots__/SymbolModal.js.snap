// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Modal
    enabled={true}
    handleClose={[Function]}
    handleOpen={[Function]}
    shortcut="symbolSearch.search.key2"
>
    <div
        className="input-wrapper"
    >
        <SearchInput
            count={0}
            handleClose={[Function]}
            handleNext={[Function]}
            handlePrev={[Function]}
            onChange={[Function]}
            onKeyUp={[Function]}
            placeholder="Search functions…"
            query=""
            size=""
            summaryMsg={undefined}
        />
    </div>
</Modal>,
  "nodes": Array [
    <Modal
      enabled={true}
      handleClose={[Function]}
      handleOpen={[Function]}
      shortcut="symbolSearch.search.key2"
>
      <div
            className="input-wrapper"
      >
            <SearchInput
                  count={0}
                  handleClose={[Function]}
                  handleNext={[Function]}
                  handlePrev={[Function]}
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  placeholder="Search functions…"
                  query=""
                  size=""
                  summaryMsg={undefined}
            />
      </div>
</Modal>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SymbolModal
        enabled={true}
        selectedSource={
                Immutable.Map {
                }
        }
        setSelectedSymbolType={[Function]}
        symbolType="functions"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SymbolModal {
        "_reactInternalInstance": [Circular],
        "buildPlaceHolder": [Function],
        "buildSummaryMsg": [Function],
        "closeModal": [Function],
        "context": Object {
          "shortcuts": undefined,
        },
        "onChange": [Function],
        "onClick": [Function],
        "onKeyUp": [Function],
        "openSymbolModal": [Function],
        "props": Object {
          "enabled": true,
          "selectedSource": Immutable.Map {
},
          "setSelectedSymbolType": [Function],
          "symbolType": "functions",
        },
        "refs": Object {},
        "renderResults": [Function],
        "selectResultItem": [Function],
        "state": Object {
          "query": "",
          "results": null,
          "resultsIndex": 0,
        },
        "traverseResults": [Function],
        "updateResults": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": Object {
        "_currentElement": <Modal
          enabled={true}
          handleClose={[Function]}
          handleOpen={[Function]}
          shortcut="symbolSearch.search.key2"
>
          <div
                    className="input-wrapper"
          >
                    <SearchInput
                              count={0}
                              handleClose={[Function]}
                              handleNext={[Function]}
                              handlePrev={[Function]}
                              onChange={[Function]}
                              onKeyUp={[Function]}
                              placeholder="Search functions…"
                              query=""
                              size=""
                              summaryMsg={undefined}
                    />
          </div>
</Modal>,
        "_debugID": 2,
        "_renderedOutput": <Modal
          enabled={true}
          handleClose={[Function]}
          handleOpen={[Function]}
          shortcut="symbolSearch.search.key2"
>
          <div
                    className="input-wrapper"
          >
                    <SearchInput
                              count={0}
                              handleClose={[Function]}
                              handleNext={[Function]}
                              handlePrev={[Function]}
                              onChange={[Function]}
                              onKeyUp={[Function]}
                              placeholder="Search functions…"
                              query=""
                              size=""
                              summaryMsg={undefined}
                    />
          </div>
</Modal>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SymbolModal
    enabled={true}
    selectedSource={
        Immutable.Map {
        }
    }
    setSelectedSymbolType={[Function]}
    symbolType="functions"
/>,
}
`;
